tests/libvmifact/factory.go

```diff
@@ -49,16 +50,19 @@ func NewFedora(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 
 // NewCirros instantiates a new CirrOS based VMI configuration
 func NewCirros(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
-	// Supplied with no user data, Cirros image takes 230s to allow login
-	withNonEmptyUserData := libvmi.WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")
-
 	cirrosOpts := []libvmi.Option{
 		libvmi.WithContainerDisk("disk0", cd.ContainerDiskFor(cd.ContainerDiskCirros)),
-		withNonEmptyUserData,
 		libvmi.WithResourceMemory(cirrosMemory()),
 	}
 	cirrosOpts = append(cirrosOpts, opts...)
-	return libvmi.New(cirrosOpts...)
+	vmi := libvmi.New(cirrosOpts...)
+
+	// Supplied with no user data, Cirros image takes 230s to allow login
+	if !cloudInitNoCloudExists(vmi.Spec.Volumes) {
+		withNonEmptyUserData := libvmi.WithCloudInitNoCloud(libvmici.WithNoCloudEncodedUserData("#!/bin/bash\necho hello\n"))
```

Comment:
this is defined twice with the same value. Do you want to make it a local variable?

Reply:
Please ignore. sorry